opal/mca/event/external/configure.m4

```diff
@@ -56,72 +56,57 @@ AC_DEFUN([MCA_opal_event_external_POST_CONFIG],[
-                 [opal_event_libdir="$with_libevent_libdir"])
-
-           opal_event_external_CPPFLAGS_save=$CPPFLAGS
+    OPAL_VAR_SCOPE_PUSH([opal_event_external_CPPFLAGS_save opal_event_external_CFLAGS_save opal_event_external_LDFLAGS_save opal_event_external_LIBS_save opal_event_dir opal_event_external_support])
+
+    # Make some processing below easier ($with_libevent==yes and
+    # $with_libevent==no has already been filtered out).
+    AS_IF([test "$with_libevent" = "external"],
+          [with_libevent=])
+
+    # Once we get to this point, $with_libevent is either: blank, a
+    # directory location, or "internal".
+
+    # Check the value of $with_libevent_libdir.  This macro safely
+    # handles "yes", "no", blank, and directory name values.
+    OPAL_CHECK_WITHDIR([libevent-libdir], [$with_libevent_libdir],
+                       [libevent.*])
+
+    # Did the user want us to check for libevent in a specific location?
+    AC_MSG_CHECKING([for external libevent in])
+    AS_IF([test -n "$with_libevent" && \
+           test "$with_libevent" != "external" && \
+           test "$with_libevent" != "internal" && \
+           test "$with_libevent" != "yes" && \
+           test "$with_libevent" != "no"],
+          [opal_event_dir=$with_libevent
+           AC_MSG_RESULT([$opal_event_dir])
+           OPAL_CHECK_WITHDIR([libevent], [$opal_event_dir],
+                              [include/event.h])
+           AS_IF([test -z "$with_libevent_libdir" || test "$with_libevent_libdir" = "yes"],
+                 [AC_MSG_CHECKING([for $with_libevent/lib64])
+                  AS_IF([test -d "$with_libevent/lib64"],
+                        [opal_event_libdir_found=yes
+                         AC_MSG_RESULT([found])],
+                        [opal_event_libdir_found=no
+                         AC_MSG_RESULT([not found])])
+                  AS_IF([test "$opal_event_libdir_found" = "yes"],
+                        [opal_event_libdir="$with_libevent/lib64"],
+                        [AC_MSG_CHECKING([for $with_libevent/lib])
+                         AS_IF([test -d "$with_libevent/lib"],
+                               [AC_MSG_RESULT([found])
+                                opal_event_libdir="$with_libevent/lib"],
+                               [AC_MSG_RESULT([not found])
+                                AC_MSG_WARN([Library directories were not found:])
+                                AC_MSG_WARN([    $with_libevent/lib64])
+                                AC_MSG_WARN([    $with_libevent/lib])
+                                AC_MSG_WARN([Please use --with-libevent-libdir to identify it.])
+                                AC_MSG_ERROR([Cannot continue])])])])],
+          [AC_MSG_RESULT([(default search paths)])])
+    AS_IF([test ! -z "$with_libevent_libdir" && test "$with_libevent_libdir" != "yes"],
+          [opal_event_libdir="$with_libevent_libdir"])
```

Comment:
Shouldn't all of the above be similar to hwloc?  I.e., shouldn't the `external/configure.m4` between hwloc and libevent be pretty similar in structure?  The exact tests will be different, of course (for various libraries, functions, header files, versions, ...etc.), but the *structure* should be similar -- especially when checking for lib vs. lib64, external-vs-non-external, ...etc.

Reply:
at first glance, my answer is yes.

But this is a revamp that is not in the scope of this PR.
Should `hwloc` or `libevent` be the reference ?